### PR TITLE
Add support for Star Tree realtime segment generation.

### PR DESCRIPTION
**Pull request by mayankshriv**
1. Added support in IndexingConfig to specify StarTreeIndexSpec.
2. Added code that picks up star tree index spec from IndexingConfig
   and passes it to segment generation.
3. Only enabled for LLC, as HLC will be phased out.
4. Added unit test for serialization/deserialization of star tree
   spec using IndexingConfig.

### CODE REVIEW

File: pinot-core/src/main/java/com/linkedin/pinot/core/data/manager/realtime/LLRealtimeSegmentDataManager.java [hunk 1]
```diff
@@ -533,7 +535,7 @@ private boolean buildSegmentInternal(boolean forCommit) {
     RealtimeSegmentConverter converter =
         new RealtimeSegmentConverter(_realtimeSegment, tempSegmentFolder.getAbsolutePath(), _schema,
             _segmentZKMetadata.getTableName(), _segmentZKMetadata.getSegmentName(), _sortedColumn,
-            _invertedIndexColumns, _noDictionaryColumns);
+            _invertedIndexColumns, _noDictionaryColumns, _starTreeIndexSpec);
```
Comment (kishoreg) [inline pinot-core/src/main/java/com/linkedin/pinot/core/data/manager/realtime/LLRealtimeSegmentDataManager.java]: this constructor has become quite big. We need to clean up this class - it can be part of another PR

File: pinot-core/src/main/java/com/linkedin/pinot/core/realtime/impl/RealtimeSegmentImpl.java [hunk 2]
```diff
@@ -695,4 +698,12 @@ public void setSegmentPartitionConfig(SegmentPartitionConfig segmentPartitionCon
   public SegmentPartitionConfig getSegmentPartitionConfig() {
     return segmentPartitionConfig;
   }
+
+  public void setStarTreeIndexSpec(StarTreeIndexSpec starTreeIndexSpec) {
```
Comment (mcvsubbu) [inline pinot-core/src/main/java/com/linkedin/pinot/core/realtime/impl/RealtimeSegmentImpl.java]: the methods to set and get start tree index spec in this class don't seem to be used. Or, did I read it wrong?

Comment (mayankshriv) [inline pinot-core/src/main/java/com/linkedin/pinot/core/realtime/impl/RealtimeSegmentImpl.java]: You are right, by the time the code was implemented these methods ended up being unused. I have removed them.
